### PR TITLE
Corrected fidelity of datetime.fromtimestamp()

### PR DESCRIPTION
**Pull request by cognitivegears**
See issue #5 for more information about the issue and steps to reproduce.

### CODE REVIEW

File: adafruit_datetime.py [hunk 1]
```diff
@@ -1284,16 +1284,18 @@ def _fromtimestamp(cls, t, utc, tz):
...
+            t = int(t)
+            us = round(frac * 1e6)
+            if us >= 1000000:
+                t += 1
+                us -= 1000000
+            elif us < 0:
+                t -= 1
+                us += 1000000
         else:
             frac = 0
```
Comment (jepler) [inline adafruit_datetime.py]: Can the assignment of `frac` be removed in the else branch?

File: adafruit_datetime.py [hunk 2]
```diff
@@ -1284,16 +1284,18 @@ def _fromtimestamp(cls, t, utc, tz):
         A timezone info object may be passed in as well.
         """
         if isinstance(t, float):
-            frac, t = _math.modf(t)
+            frac = t - int(t)
+            t = int(t)
```
Comment (jepler) [inline adafruit_datetime.py]: I'm not sure whether this is more efficient than modf, but perhaps you determined it was.

**Review by jepler — CHANGES_REQUESTED** [review]
Thanks! It looks like one small clean-up is possible, let me know if I'm mistaken about it.

File: adafruit_datetime.py (hunk 1 above)
Comment (cognitivegears) [reply adafruit_datetime.py]: Removed, thanks!

File: adafruit_datetime.py (hunk 2 above)
Comment (cognitivegears) [reply adafruit_datetime.py]: Reverted - this was part of testing to see if I could also resolve the timestamp on the fractional case, but was not effective. If we got rid of all of them we could get rid of a small amount of memory by removing the import on the math module, but doesn't seem worth it.

**Review by jepler — APPROVED** [review]
Thank you!  The Actions failures when linting the code can be ignored for the purposes of this PR.

**Pull request merged**